### ISSUE TITLE
Explain: missing variable #2 while planning registers (exception location: /usr/src/packages/BUILD/arangod/Aql/ExecutionNode.cpp:1163). Please report this error to arangodb.com (while finalizing)

### ISSUE THREAD

**Issue by amurchick** [arangodb/arangodb#1970]
Arango error
```
Explain: missing variable #2 while planning registers (exception location: /usr/src/packages/BUILD/arangod/Aql/ExecutionNode.cpp:1163). Please report this error to arangodb.com (while finalizing)
```

ArangoDb 3.0.3

## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.0
- [ ] self-compiled devel branch

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [x] Linux 
  - [x] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:

### this is an AQL-related issue:
[ ] I'm using graph features

I'm issuing AQL via:
- [x] web interface with this browser: Chrome/52.0.2743.82 Safari/537.36 running on this OS: Intel Mac OS X 10_11_6
- [ ] arangosh
- [ ] this Driver:

I've run `db._explain("<my aql query>")` and it didn't shed more light on this.
The AQL query in question is:
```
LET emails = (RETURN ["any@host.email","any1@host.email"])
FOR e IN emails
    FOR u IN users
        FILTER e.email IN u.emails[*].value
        LET matches = (
            FOR i IN u.emails
                FOR email IN emails
                    FILTER email.email == i.value
                    RETURN DISTINCT {
                        what: email,
                        where: i
                    }
        )
        RETURN {
            _key: u._key,
            name: u.name,
            emails: u.emails,
            matches
        }
```
Note: query with errors, only for error reproduce.

The issue can be reproduced using this dataset:
```
on any empty collection
```
Please provide a way to create the dataset to run the above query on; either by a gist with an arangodump, or `db.collection.save({my: "values"}) statements. If it can be reproduced with one of the ArangoDB example datasets, it's a plus.

### this is a web interface-related issue:
I'm using the web interface with this browser:     running on this OS:
- [x] authentication is enabled?
- [ ] using the cluster?
- [ ] _system database?

These are the steps to reproduce:
1) open the browser on http://127.0.0.1:8529
2) log in as root
3) use database [ ] `_system` [x] other: any database
4) paste query to AQL editor and press "Explain"

The following problem occurs: [Screenshot?] 
Instead I would be expecting: 

### this is an installation-related issue:
Describe which steps you carried out, what you expected to happen and what actually happened.

**Comment by jsteemann** [comment]
I tried to reproduce this, but the query works in the current `3.0` branch. Will now check with latest release 3.0.3.

**Comment by jsteemann** [comment]
Probably related to https://github.com/arangodb/arangodb/issues/1950.

**Comment by jsteemann** [comment]
The error occurs in 3.0.3, and was fixed in the 3.0 branch after the release. The root cause is the same as for https://github.com/arangodb/arangodb/issues/1950.

**Comment by jsteemann** [comment]
Fixed in 3.0.4

**Comment by jsteemann** [comment]
3.0.4 released today.
